src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java

```diff
@@ -0,0 +1,113 @@
+package uk.gov.hmcts.reform.cwrdapi.service;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.cwrdapi.client.domain.CaseWorkerDomain;
+import uk.gov.hmcts.reform.cwrdapi.client.domain.CaseWorkerProfile;
+import uk.gov.hmcts.reform.cwrdapi.client.domain.Location;
+import uk.gov.hmcts.reform.cwrdapi.client.domain.Role;
+import uk.gov.hmcts.reform.cwrdapi.client.domain.WorkArea;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.CaseWorkerLocationRequest;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.CaseWorkerRoleRequest;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.CaseWorkerWorkAreaRequest;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.CaseWorkersProfileCreationRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class CaseWorkerProfileConverter implements Converter<List<CaseWorkerDomain>,
+        List<CaseWorkersProfileCreationRequest>> {
+    public static final String COMMA = ",";
+
+    /**
+     * Convert the source object of type {@code CaseWorkerDomain}
+     * to target type {@code CaseWorkersProfileCreationRequest}.
+     *
+     * @param caseWorkerProfiles the source object to convert,
+     *                           which must be an instance of {@code CaseWorkerDomain} (never {@code null})
+     * @return the converted object, which must be an instance of {@code CaseWorkersProfileCreationRequest}
+     *              (potentially {@code null})
+     * @throws IllegalArgumentException if the source cannot be converted to the desired target type
+     */
+    public List<CaseWorkersProfileCreationRequest> convert(List<CaseWorkerDomain> caseWorkerProfiles) {
+        List<CaseWorkersProfileCreationRequest> caseWorkersProfileCreationRequests = new ArrayList<>();
+        caseWorkerProfiles
```

Comment:
ofNullable().ifPresent may validate else null pointer may get

Reply:
we are checking whether the list is empty or not before calling convert in my latest change

Reply:
ok